CppSamples/EditData/EditGeodatabaseWithTransactions/EditGeodatabaseWithTransactions.cpp

```diff
@@ -0,0 +1,613 @@
+}
+
+void EditGeodatabaseWithTransactions::rollbackTransaction()
+{
+  if (!m_geodatabase)
+  {
+    return;
+  }
+
+  // Rollback the transaction to discard the edits (this will also end the transaction)
+  m_geodatabase->rollbackTransaction();
+  setMessageText("Edits discarded.");
+}
+
+void EditGeodatabaseWithTransactions::cancelTransaction()
+{
+  // User canceled - keep transaction active
+  setMessageText("Transaction started.");
+}
+
+void EditGeodatabaseWithTransactions::updateFeatureTypesForTable(GeodatabaseFeatureTable *table)
+{
+  if (!table)
+  {
+    return;
+  }
+
+  m_currentFeatureTypes.clear();
+
+  QStringList featureTypeNames;
+  for (const FeatureType &featureType : table->featureTypes())
+  {
+    featureTypeNames.append(featureType.name());
+  }
+
+  featureTypeNames.sort(Qt::CaseInsensitive);
+  m_currentFeatureTypes = featureTypeNames;
+
+  emit currentFeatureTypesChanged();
+}
+
+void EditGeodatabaseWithTransactions::setSelectedTableName(const QString &tableName)
+{
+  if (m_selectedTableName != tableName)
+  {
+    m_selectedTableName = tableName;
+    emit selectedTableNameChanged();
+
+    // Update feature types for the newly selected table
+    GeodatabaseFeatureTable *table = m_tablesByName.value(tableName, nullptr);
+    if (table)
+    {
+      updateFeatureTypesForTable(table);
+    }
+  }
+}
+
+void EditGeodatabaseWithTransactions::addFeatureAtLocation(const QString &tableName, const QString &featureTypeName, const QPoint &location)
+{
+  if (!m_mapView || !m_geodatabase)
+  {
+    return;
+  }
+
+  // Get the correct table using dynamic lookup
+  GeodatabaseFeatureTable *targetTable = m_tablesByName.value(tableName, nullptr);
+
+  if (!targetTable)
+  {
+    setMessageText(QString("Table '%1' not found.").arg(tableName));
+    return;
+  }
+
+  // Find the feature type
+  FeatureType selectedFeatureType;
+  bool foundType = false;
+  for (const FeatureType &featureType : targetTable->featureTypes())
+  {
+    if (featureType.name() == featureTypeName)
+    {
+      selectedFeatureType = featureType;
+      foundType = true;
+      break;
+    }
+  }
+
+  if (!foundType)
+  {
+    setMessageText(QString("Feature type '%1' not found in table '%2'.").arg(featureTypeName, tableName));
+    return;
+  }
+
+  // Convert screen point to map point
+  Point mapPoint = m_mapView->screenToLocation(location.x(), location.y());
+
+  if (mapPoint.isEmpty())
+  {
+    setMessageText("Invalid location clicked.");
+    return;
+  }
+
+  // Convert point to WGS84 for comparison with extent (extent is in WGS84)
+  Point wgs84Point = mapPoint;
+  if (!mapPoint.spatialReference().isEmpty() && mapPoint.spatialReference().wkid() != 4326)
+  {
+    // Project the point to WGS84
+    Geometry projectedGeometry = GeometryEngine::project(mapPoint, SpatialReference::wgs84());
+    if (!projectedGeometry.isEmpty())
+    {
+      wgs84Point = static_cast<Point>(projectedGeometry);
+    }
```

Comment:
You didn't go with my suggestion to use geometry_cast. That is what we recommend for our users, so we should do the same. If you want to use a different pattern, please let me know why.

https://github.com/Esri/arcgis-maps-sdk-samples-qt/pull/1848#discussion_r2323714947